client.go

```diff
@@ -100,12 +110,18 @@ func (cli *Client) sendRequest(method string, url *url.URL, body io.Reader, head
 	req.Header.Set("Accept-Encoding", "gzip")
 	req.Header.Set("X-Apikey", cli.APIKey)
 
-	if headers != nil {
-		for k, v := range headers {
-			req.Header.Set(k, v)
+	if headers == nil {
```

Comment:
I find this a bit problematic. The `sendRequest` function should receive the headers in the `headers` argument, or take them from the `Client` object, but not both. Merging the headers in the arguments with the headers in the object can result in some header being overwritten with a different value. In this case the value in the object is taking precedence, which is counter-intuitive, because if you explicitly pass header `Foo=Bar` as an argument to `sendRequest` you won't expect that request is done with `Foo=Baz`. 

If we decide merging both sets of headers it should be the other way around, the arguments to `sendRequest` has precedence,  and that behaviour should be clearly documented. I lean towards not merging them and using only the arguments.

Reply:
That's a good point, definitively global headers shouldn't override headers in sendRequest arguments. Your last suggestion, not merging, make sense but raise some doubts, in some place we need to reconcile "global" headers with the ones passed directly to the methods. I mean, we can go a level up, Post, Get, etc... bu the problem is still there, how reconcile global defined headers with local ones.

Reply:
Yep, I wrote the comment before realising that `WithHeader` is used at the individual request level, I was assuming that all headers were set at the client level. As we have both, we must surely merge the headers at the client with those specified for individual requests, but headers at the request level must override headers at the client level in case of collision.

Reply:
I also find a bit confusing that at the request level you specify each individual header with a different `WithHeader` argument, while at the client level you use a single `WithHeaders` argument that receives a map. We can't have two functions named `WithHeader`, one for requests and the other for clients, but maybe we have have `WithHeader` for requests, and `WithGlobalHeader` or `WithHTTPHeader` for clients.

My point is that both should be singular or both plural, or we should have both variants for requests and clients.  For example..

For requests: `WithHeader` (already exits) and `WithHeaders` (a variant that receives a map)
For clients: `WithGlobalHeader` (individual header) and `WithGlobalHeaders` (map with headers)

However I wouldn't keep things simple and have only the singular variant for each.